### ISSUE TITLE
Expired Accounts - Teachers can still login

### ISSUE THREAD

**Issue by mnichols13** [MyCaert/Issues-Enhancements#410]
@bguerrettaz  @BrianPeterlin @jquesse2  refer to Github 338
Teachers in all states that expired can still open their home page.  This allows them to access all of the My Community items which are under copyright.  Please provide an estimate.  This needs to be resolved asap.  We have had this issue multiple times, I am not sure why we have to continue to pay for this.   For example **Username**: LemmonAG **Password**: 57638AG

**Comment by bguerrettaz** [comment]
Could you give me a few examples that I can test with to see how they're
getting through and what it would take to fill the gap please?

**Comment by mnichols13** [comment]
In the original message:  
Username: LemmonAG Password: 57638AG

**Comment by mnichols13** [comment]
![image](https://github.com/user-attachments/assets/29eb116d-7f7c-4d9a-9ff2-2e329d510ce5)

**Comment by mnichols13** [comment]
I have went into some of these and manually changed them to inactive, but not all of them.

**Comment by mnichols13** [comment]
@bguerrettaz  - pls update me on when you can fix this


Delaware: 
Christiana HS 
THOMAS.LUNDY@christina.k12.de.us

IL
Nashville
jbond@nashville-k12.org

**Comment by bguerrettaz** [comment]
Ok, I dug into the full process of login, status checks, date checks, etc.  There are a few scheduled jobs that don't seem to be running.  I manually ran them today and that resolved the issue you're seeing.  For this issue to be resolved, we need to make sure that these jobs are scheduled weekly (the .bat files in this screenshot):

<img width="641" alt="image" src="https://github.com/user-attachments/assets/cb3fc44d-706d-4eaa-b50c-51005510baae">

@BrianPeterlin - can you assist with this?  I thought it was scheduled in job scheduler but it seems to be gone now and I'm not sure how well those run?  Thoughts?

So to be clear, this is resolved as of today but we need to make sure these scripts run to update the database records to avoid teachers logging in when they shouldn't have access to.

**Comment by mnichols13** [comment]
@bguerrettaz @BrianPeterlin @jquesse2  @LKocher  This seems to be a recurring conversation.  Why can't we get these jobs to run consistently?

**Comment by mnichols13** [comment]
@BrianPeterlin @bguerrettaz   Please run this job this weekend.

**Comment by BrianPeterlin** [comment]
@mnichols13 , try now

**Comment by BrianPeterlin** [comment]
@bguerrettaz, which job is appearing not to run?  There was a setting on the job to run even when the user is not logged in.  I have enabled that in the Windows Scheduler.

![image](https://github.com/user-attachments/assets/184b04be-a625-40af-9714-c796c1f869ac)

**Comment by mnichols13** [comment]
yes it is fixed.   Can this job be run on Dec 2 ? @bguerrettaz @BrianPeterlin

**Comment by BrianPeterlin** [comment]
It is set up to run every Sunday, but we will need to keep an eye on it and
make sure that it executes properly

Brian Peterlin
***@***.***
Cell# 309 310 4551

**Comment by mnichols13** [comment]
@BrianPeterlin @bguerrettaz  Report did not run and execute properly  Teachers can still login to their accounts even when the master deactivate date is 12-1-24...  Please re-run / fix ASAP

**Comment by bguerrettaz** [comment]
@Brian Peterlin ***@***.***> - I think you scheduled this?

**Comment by BrianPeterlin** [comment]
try the teachers now

Brian Peterlin
***@***.***
Cell# 309 310 4551

**Comment by mnichols13** [comment]
fixed

**Comment by mnichols13** [comment]
@BrianPeterlin Are you going to run this job?  Why can't this be set up to run regularly around the 2nd day of every month?

**Comment by mnichols13** [comment]
@BrianPeterlin 
Please schedule this job for these days:
February 2
March 2
April 6
May 4
June 1
July 6
Aug 3
Sept 7
Oct 5
Nov 2

**Comment by mnichols13** [comment]
@BrianPeterlin @jquesse2 This report did not get run as requested yesterday, Feb 2nd.  Why?

**Comment by mnichols13** [comment]
This teacher should not be able to login, he expired on 2-1-25

![Image](https://github.com/user-attachments/assets/a5f8d75a-75a2-4f8a-9fbd-1cf546daccba)

**Comment by BrianPeterlin** [comment]
try this URL and see if it expires the account

http://www.mycaert.com/ScheduledJobs/ExpiringAccounts.asp


Brian Peterlin
***@***.***
Cell# 309 310 4551

**Comment by mnichols13** [comment]
this is what I get 

![Image](https://github.com/user-attachments/assets/c2ad8fd4-7468-4c12-a30a-25627139a5c3)

**Comment by BrianPeterlin** [comment]
did it expire account?

Brian Peterlin
***@***.***
Cell# 309 310 4551

**Comment by mnichols13** [comment]
nothing changed

**Comment by BrianPeterlin** [comment]
all jobs have now been manually run

Brian Peterlin
***@***.***
Cell# 309 310 4551

**Comment by BrianPeterlin** [comment]
all good?

Brian Peterlin
***@***.***
Cell# 309 310 4551

**Comment by mnichols13** [comment]
@BrianPeterlin can we please run the necessary job this weekend to ensure the teachers with expired accounts on March 1 cannot login.

**Comment by BrianPeterlin** [comment]
complete

Brian Peterlin
***@***.***
Cell# 309 310 4551

**Comment by mnichols13** [comment]
@BrianPeterlin job for May 4 was not run.  Teachers can still login with expiration dates of 5-1-2025 Please take care of ASAP

**Comment by mnichols13** [comment]
@BrianPeterlin can this report be run over the weekend?  Sunday June 1 ?

**Comment by bguerrettaz** [comment]
@Brian Peterlin ***@***.***>

**Comment by mnichols13** [comment]
@BrianPeterlin @jquesse2 @LKocher 

Job was not run over the weekend as requested.  When can this run?
